core/src/com/unciv/ui/screens/newgamescreen/NewGameScreen.kt

```diff
@@ -354,11 +358,37 @@ class NewGameScreen(
         }
     }
 
-    fun updateRuleset() {
+    /** Updates our local [ruleset] from [gameSetupInfo], guarding against exceptions.
+     *
+     *  Note: The options reset on failure is not propagated automatically to the Widgets -
+     *  the caller must ensure that.
+     *
+     *  @return Success - failure means gameSetupInfo was reset to defaults and the Ruleset was reverted to G&K
+     */
+    fun updateRuleset(): Boolean {
+        var success = true
+        fun handleFailure(message: String): Ruleset {
+            success = false
+            gameSetupInfo = GameSetupInfo()
+            ToastPopup(message, this, 5000)
+            return RulesetCache[BaseRuleset.Civ_V_GnK.fullName]!!
```

Comment:
Why are we resetting to GnK? I would expect us to go back to the previously selected ruleset, not to default